### PR TITLE
added events code around duplicates

### PR DESCRIPTION
**Pull request by Code0987**

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -3565,4 +3583,14 @@ export class AddEditExpensePage implements OnInit {
       this.trackingService.viewComment({ Asset: 'Mobile' });
     }
   }
+
+  setDuplicateBoxOpen(value) {
+    this.duplicateBoxOpen = value;
+
+    if (value) {
+      this.trackingService.duplicateDetectionUserActionExpand({ Asset: 'Mobile', Page: this.mode === 'add' ? 'Add Expense' : 'Edit Expense' });
```
Comment (pradeep8093) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3591]: remove extra space at the end

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -2243,4 +2261,14 @@ export class AddEditMileagePage implements OnInit {
       this.trackingService.viewComment({ Asset: 'Mobile' });
     }
   }
+
+  setDuplicateBoxOpen(value) {
+    this.duplicateBoxOpen = value;
+
+    if (value) {
+      this.trackingService.duplicateDetectionUserActionExpand({ Asset: 'Mobile', Page: this.mode === 'add' ? 'Add Mileage' : 'Edit Mielage' });
```
Comment (pradeep8093) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2269]: remove extra space

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 3]
```diff
@@ -2085,5 +2103,15 @@ export class AddEditPerDiemPage implements OnInit {
       this.trackingService.viewComment({ Asset: 'Mobile' });
     }
   }
+
+  setDuplicateBoxOpen(value) {
+    this.duplicateBoxOpen = value;
+
+    if (value) {
+      this.trackingService.duplicateDetectionUserActionExpand({ Asset: 'Mobile', Page: this.mode === 'add' ? 'Add Per Diem' : 'Edit Per Diem' });
```
Comment (pradeep8093) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:2111]: remove extra space

**Comment by pradeep8093** [comment]
also shouldn’t we track

User name, 
User org name
User org Id

while tracking events

**Comment by Code0987** [comment]
> also shouldn’t we track
> 
> User name,
> User org name
> User org Id
> 
> while tracking events

Updated

**Review by pradeep8093 — APPROVED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts [hunk 4]
```diff
@@ -335,6 +335,23 @@ export class AddEditPerDiemPage implements OnInit {
         setTimeout(() => {
           this.pointToDuplicates = false;
         }, 3000);
+
+        this.etxn$.pipe(take(1)).subscribe(async etxn => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts:339]: extract to a function

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 5]
```diff
@@ -398,6 +398,23 @@ export class AddEditMileagePage implements OnInit {
         setTimeout(() => {
           this.pointToDuplicates = false;
         }, 3000);
+
+        this.etxn$.pipe(take(1)).subscribe(async etxn => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:402]: extract to a function

**Review by Julias0 — APPROVED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts (hunk 4 above)
Comment (Code0987) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem.page.ts]: done

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 5 above)
Comment (Code0987) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts]: done

**Pull request merged**